src/state/queries/handle.ts

```diff
@@ -0,0 +1,27 @@
+import React from 'react'
+import {useQueryClient} from '@tanstack/react-query'
+
+import {useSession} from '#/state/session'
+
+const fetchHandleQueryKey = (handleOrDid: string) => ['handle', handleOrDid]
+
+export function useFetchHandle() {
+  const {agent} = useSession()
+  const queryClient = useQueryClient()
+
+  return React.useCallback(
+    async (handleOrDid: string) => {
+      if (handleOrDid.startsWith('did:')) {
+        const res = await queryClient.fetchQuery({
+          // cache in memory forever, page reload clears
+          staleTime: Infinity,
```

Comment:
@pfrazee

Reply:
we should invalidate the cache when the user changes their handle, and might want to manually update the cache any time we use the get profile hook